tests/test_record_processor.py

```diff
@@ -0,0 +1,188 @@
+            },
+            {
+                "recordId": "456",
+                "result": "Ok",
+                "data": "eyJwYXRyb25faWQiOiAiYWJjMTIzNDU2Nzg5MTAxMTEyMTMxNDE1IiwgInBhdHJvbl9yZXRyaWV2YWxfc3RhdHVzIjogIm1pc3NpbmciLCAicHR5cGVfY29kZSI6IG51bGwsICJwYXRyb25faG9tZV9saWJyYXJ5X2NvZGUiOiBudWxsLCAicGNvZGUzIjogbnVsbCwgInBvc3RhbF9jb2RlIjogbnVsbCwgImdlb2lkIjogbnVsbCwgImtleSI6ICIxMjQiLCAibWludXRlc191c2VkIjogMTIsICJ0cmFuc2FjdGlvbl9ldCI6ICIyMDIxLTEyLTE0IiwgImJyYW5jaCI6ICJicmFuY2ggZGVmIiwgImFyZWEiOiAiYXJlYSAyIiwgInN0YWZmX292ZXJyaWRlIjogbnVsbH0=",
+            },
+            {
+                "recordId": "789",
+                "result": "Ok",
+                "data": "eyJwYXRyb25faWQiOiAiYWJjMTIzNDU2Nzg5MTAxMTEyMTMxNDE1MTYiLCAicGF0cm9uX3JldHJpZXZhbF9zdGF0dXMiOiAiZ3Vlc3QgcGFzcyIsICJwdHlwZV9jb2RlIjogbnVsbCwgInBhdHJvbl9ob21lX2xpYnJhcnlfY29kZSI6IG51bGwsICJwY29kZTMiOiBudWxsLCAicG9zdGFsX2NvZGUiOiBudWxsLCAiZ2VvaWQiOiBudWxsLCAia2V5IjogIjEyNSIsICJtaW51dGVzX3VzZWQiOiAxNSwgInRyYW5zYWN0aW9uX2V0IjogIjIwMjEtMTItMTUiLCAiYnJhbmNoIjogImJyYW5jaCBnaGkiLCAiYXJlYSI6ICJhcmVhIDMiLCAic3RhZmZfb3ZlcnJpZGUiOiBudWxsfQ==",
+            },
+        ]
+        for i, record in enumerate(valid_records):
+            assert (
+                test_pc_reserve_instance.process_record(record, "json")
+                == expected_processed_records[i]
+            )
+
+    def test_process_records_csv_no_exception(
+        self, test_data, test_location_hours_instance, caplog
+    ):
+        valid_records = test_data["valid_records_csv"]["records"]
+        expected_processed_records = [
+            {
+                "recordId": "49546986683135544286507457936321625675700192471156785154",
+                "result": "Ok",
+                "data": "YWF8TGlicmFyeSBBfE1vbnwwOTowMHwxNzowMHwyMDIzLTAxLTAxCg==",
+            },
+            {
+                "recordId": "49546986683135544286507457936321625675700192471156785154",
+                "result": "Ok",
+                "data": "YmJ8TGlicmFyeSBCfFR1ZXx8fDIwMjMtMDItMDIK",
+            },
+            {
+                "recordId": "49546986683135544286507457936321625675700192471156785154",
+                "result": "Ok",
+                "data": "Y2N8TGlicmFyeSBcfCBDICEsPXxXZWR8MDE6MDB8MjM6MDB8MjAyMy0wMy0wMwo=",
+            },
+        ]
+
+        for i, record in enumerate(valid_records):
+            assert (
+                test_location_hours_instance.process_record(record, "csv")
+                == expected_processed_records[i]
+            )
+
+    def test_process_records_exception(
+        self, test_data, test_pc_reserve_instance, test_location_hours_instance
+    ):
+        invalid_records = test_data["invalid_records"]
+        with pytest.raises(Exception):
```

Comment:
split this up into two `with pytest.raises(Exception)` clauses so that we test that both of them thrown an exception